src/client/options/mod.rs

```diff
@@ -1325,6 +1325,19 @@ impl ClientOptions {
             }
         }
 
+        if let Some(heartbeat_frequency) = self.heartbeat_freq {
```

Comment:
This validation existed, but it was in the URI parsing stage. I've moved it to `ClientOptions::validate` to ensure it gets hit even when the options are constructed directly.